File: TaskShop/TaskShop/Scripts/Models/SearchModel.js
var SearchForm = function (options) {
    var self = this;

    self.Model = ko.observable();
    self.Producer = ko.observable();
    self.GoodsCategory = ko.observable();
    self.MaxPrice = ko.observable();
    self.MinPrice = ko.observable();
};

var SearchViewModel = function (options) {
    var self = this;


    self.SearchForm = new SearchForm();
    self.GoodsCategories = ko.observableArray(["All", "Monitors", "Batteries", "MemoryCards", "Earphones"]);


    self.arr = ko.observableArray([]);

    self.standartMode = ko.observable(true);
    self.fullMode = ko.computed(function () {
        return !(self.standartMode());
    }, self);


    self.Full = function (data) {
        self.standartMode(false);
    };
    self.Cancel = function (data) {
        self.standartMode(true);
    };


    self.GetAll = function (data) {
        var json = ko.toJS(self.SearchForm);
        $.post(options.searchGetAll, json, function (returnedData) {
            if (returnedData)
                self.arr(returnedData);
        });
    };
}

File: TaskShop/DataLayer/Models/SearchModel.cs
using System;

namespace DataLayer.Models
{
    public class SearchModel
    {
        public string GoodsCategory { get; set; }
        public String Producer { get; set; }
        public String Model { get; set; }
        public int MinPrice { get; set; }
        public int MaxPrice { get; set; }
    }

    public class ReturnSearchModel
    {
        public int GoodsId { get; set; }

        public string GoodsCategory { get; set; }

        public String Producer { get; set; }

        public String Model { get; set; }

        public int Price { get; set; }
    }
}